### PR TITLE
Refactor signal handling for FreeBSD ##r_debug

### PR DESCRIPTION
**Pull request by devnexen**
Solving r_debug_handle_signals build warning issue.

### CODE REVIEW

File: libr/debug/p/debug_native.c [hunk 1]
```diff
@@ -110,6 +110,39 @@ static int r_debug_handle_signals(RDebug *dbg) {
...
+			return 0;
+		}
+	}
+
+	siginfo = linfo.pl_siginfo;
+	dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+	dbg->reason.signum = siginfo.si_signo;
+
+	switch (dbg->reason.signum) {
+		case SIGABRT:
```
Comment (radare) [inline libr/debug/p/debug_native.c]: Bad indent

File: libr/debug/p/debug_native.c [hunk 2]
```diff
@@ -110,6 +110,39 @@ static int r_debug_handle_signals(RDebug *dbg) {
...
+	siginfo_t siginfo;
+	int ret = ptrace (PT_LWPINFO, dbg->pid, (char *)&linfo, sizeof (linfo));
+	if (ret == -1) {
+		if (errno == ESRCH) {
+			dbg->reason.type = R_DEBUG_REASON_DEAD;
+			return 0;
+		}
+		r_sys_perror ("ptrace PTRACE_LWPINFO");
+		return -1;
+	} else {
```
Comment (radare) [inline libr/debug/p/debug_native.c]: No need to else after return

File: libr/debug/p/debug_native.c (hunk 1 above)
Comment (devnexen) [reply libr/debug/p/debug_native.c]: You sure ? seems similar to other switch/cases.

File: libr/debug/p/debug_native.c (hunk 2 above)
Comment (devnexen) [reply libr/debug/p/debug_native.c]: true

**Pull request merged**